skt/state_file.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python2
+
+# Copyright (c) 2017 Red Hat, Inc. All rights reserved. This copyrighted
+# material is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General
+# Public License v.2 or later.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+"""Functions that manage the skt state file."""
+import logging
+import os
+import yaml
+
+
+def destroy(cfg):
+    """
+    Destroy the state file.
+
+    Args:
+        cfg: A dictionary of skt configuration
+    """
+    state_file = cfg.get('state')
+
+    # Check if the state file exists
+    if os.path.isfile(state_file):
+
+        # Is the state file writable?
+        if not os.access(state_file, os.W_OK):
+            raise IOError("Cannot delete state file: {}".format(state_file))
+
+        # Delete the state file
+        os.unlink(state_file)
+
+
+def read(cfg):
+    """
+    Read the state file from the disk.
+
+    Args:
+        cfg: A dictionary of skt configuration
+    """
+    # Return an empty state dictionary if the state file does not exist
+    current_state = {}
+
+    state_file = cfg.get('state')
+    logging.debug("Reading state from: %s", state_file)
+
+    # Check if the state file exists
+    if os.path.isfile(state_file):
+
+        # Is the state file readable?
+        if not os.access(state_file, os.R_OK):
+            raise IOError("Cannot read state file: {}".format(state_file))
+
+        # Read from the state file
+        with open(state_file, 'r') as fileh:
+            current_state = yaml.load(fileh)
+
+    return current_state
+
+
+def update(cfg, state_updates):
+    """
+    Save state to the state file on disk.
```

Comment:
This would be nice to update to "update" :)

Reply:
Oops. Thanks.